api/machine/machine.proto

```diff
@@ -141,9 +141,15 @@ message Event {
 }
 
 // rpc reset
+message ResetPartitionSpec {
+  string label = 1;
+  bool wipe = 2;
+}
+
 message ResetRequest {
   bool graceful = 1;
   bool reboot = 2;
+  repeated ResetPartitionSpec system_disk = 3;
```

Comment:
The reason is not explained here, and that's my miss (need to add comment).

Idea is that right now we only allow resetting (wiping) system disk partitions, but once we grow to support labels for user disks, we can extend this API to accept another list of user disks/partitions to wipe. So this field is the beginning of the future which applies to system disk only.

Reply:
I can understand that; the issue is that _changing_ a protobuf spec is hard/disruptive, where _adding_ is not.  So it makes more sense to define more generic option names (`targets` instead `system_disk`) instead of changing them later, just because the scope changes.

Reply:
This makes sense to me. Does a more generic name not work, Andrey? Just trying to understand the motivation for `system` here.

Reply:
I don't quite get this.

Today we have:

```
systemDisk []ResetPartitionSpec
```

then we add to it:

```
userDisks map[device][]ResetPartitionSpec
```

So there is nothing to be removed. We just don't enforce users to name system disk as `/dev/sda` (or anything else it is), as it's special and Talos knows where the system disk is.

Reply:
settled on `system_partitions_to_wipe`